usi/src/main/java/daq/usi/ovs/OpenVSwitch.java

```diff
@@ -1,35 +1,57 @@
 package daq.usi.ovs;
 
-import daq.usi.ISwitchController;
 import daq.usi.ResponseHandler;
-import grpc.Interface;
+import daq.usi.SwitchController;
+import grpc.InterfaceResponse;
 import grpc.LinkStatus;
 import grpc.POEStatus;
 import grpc.POESupport;
-import grpc.Power;
+import grpc.PowerResponse;
 import grpc.SwitchActionResponse;
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.net.URL;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
-public class OpenVSwitch implements ISwitchController {
-  private String fauxIface;
+public class OpenVSwitch implements SwitchController {
 
-  public OpenVSwitch(String fauxIface) {
-    this.fauxIface = fauxIface;
+  private static final String OVS_OUTPUT_FILE = "ovs_output.txt";
+
+  protected String getInterfaceByPort(int devicePort) throws FileNotFoundException {
+    URL file = OpenVSwitch.class.getClassLoader().getResource(OVS_OUTPUT_FILE);
```

Comment:
why is this a getResource -- why not just reading a file from the disk? At least comment the why here -- this doesn't match my expectations.

Reply:
getResource is reading from a file but this way makes the files’ path independent. It also makes Unit testing easier where it automatically uses a difference resources folder.